R/imageutils.R

```diff
@@ -4,14 +4,50 @@ startPNG <- function(filename, width, height, res, ...) {
   # Finally, if neither quartz nor Cairo, use png().
   if (capabilities("aqua")) {
     pngfun <- grDevices::png
+  } else if ((getOption('shiny.useragg') %OR% FALSE) &&
+             nchar(system.file(package = "ragg"))) {
+    # ragg seems preferrable to Cairo, especially when it comes to custom fonts
+    # https://github.com/yixuan/showtext/issues/33#issuecomment-620848077
+    # https://ragg.r-lib.org/articles/ragg_quality.html
+    pngfun <- ragg::agg_png
   } else if ((getOption('shiny.usecairo') %OR% TRUE) &&
-      nchar(system.file(package = "Cairo"))) {
+             nchar(system.file(package = "Cairo"))) {
     pngfun <- Cairo::CairoPNG
   } else {
     pngfun <- grDevices::png
   }
 
-  pngfun(filename=filename, width=width, height=height, res=res, ...)
+  args <- rlang::list2(filename=filename, width=width, height=height, res=res, ...)
+
+  # The order of priority for the bg color is:
+  # (1) User specified bg via `renderPlot()`
+  # (2) bg option was set via thematic
+  if (is.null(args$bg) && rlang::is_installed("thematic")) {
+    # TODO: use :: once thematic is on CRAN
+    args$bg <- getFromNamespace("thematic_get_option", "thematic")("bg", "white")
+    # auto vals aren't resolved until plot time, so if we see one, resolve it
+    if (isTRUE("auto" == args$bg)) {
+      args$bg <- getCurrentOutputInfo()[["bg"]]
+    }
+  }
+
+  # Handle both bg and background device arg
+  # https://github.com/r-lib/ragg/issues/35
+  fmls <- names(formals(pngfun))
+  if (("background" %in% fmls) && (!"bg" %in% fmls)) {
+    if (is.null(args$background)) {
+      args$background <- args$bg
+    }
+    args$bg <- NULL
+  }
```

Comment:
Maybe scope to the ragg case?

Reply:
Decided against this...could be useful if and when we support a general device function